client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt

```diff
@@ -0,0 +1,414 @@
+import io.spine.examples.chatspn.account.event.AccountNotCreated
+import io.spine.examples.chatspn.chat.ChatPreview
+import io.spine.examples.chatspn.chat.command.CreatePersonalChat
+import io.spine.examples.chatspn.message.MessageView
+import io.spine.examples.chatspn.message.command.SendMessage
+import io.spine.examples.chatspn.message.event.MessageMarkedAsDeleted
+import io.spine.net.EmailAddress
+
+/**
+ * Provides API to communicate with ChatSpn server.
+ */
+public class ClientFacade {
+    public var authenticatedUser: UserProfile? = null
+    private val client: Client
+    private val userChatsSubscriptions = mutableListOf<Subscription>()
+    private val messagesSubscriptions = mutableListOf<Subscription>()
+
+    init {
+        val channel = ManagedChannelBuilder.forAddress(
+            "localhost",
+            DEFAULT_CLIENT_SERVICE_PORT
+        )
+            .usePlaintext()
+            .build()
+        client = Client.usingChannel(channel).build()
+    }
+
+    /**
+     * Registers a new user.
+     *
+     * @param name name of the user to register
+     * @param email email of the user to register
+     * @param onSuccess will be called when the user successfully passed registration
+     * @param onFail will be called when the user failed the registration
+     */
+    public fun register(
+        name: String,
+        email: String,
+        onSuccess: () -> Unit = {},
+        onFail: () -> Unit = {}
+    ) {
+        val command = createAccount(name, email)
+
+        var successSubscription: Subscription? = null
+        var failSubscription: Subscription? = null
+        successSubscription = subscribeToEvent(
+            command.id,
+            AccountCreated::class.java
+        ) { event ->
+            cancelSubscription(successSubscription!!)
+            cancelSubscription(failSubscription!!)
+            authenticatedUser = findUser(event.user)
+            onSuccess()
+        }
+        failSubscription = subscribeToEvent(
+            command.id,
+            AccountNotCreated::class.java
+        ) { event ->
+            cancelSubscription(successSubscription)
+            cancelSubscription(failSubscription!!)
+            onFail()
+        }
+        clientRequest()
+            .command(command)
+            .postAndForget()
+    }
+
+    /**
+     * Authenticates the user.
+     *
+     * @param email email of the user to authenticate
+     * @param onSuccess will be called when the user successfully passed the authentication
+     * @param onFail will be called when the user failed the authentication
+     */
+    public fun logIn(
+        email: String,
+        onSuccess: () -> Unit = {},
+        onFail: () -> Unit = {}
+    ) {
+        val user = findUser(email)
+        if (null == user) {
+            onFail()
+        } else {
+            authenticatedUser = user
+            onSuccess()
+        }
+    }
+
+    /**
+     * Finds user by ID.
```

Comment:
Please include the parameter descriptions and return type descriptions into each method documented currently.